### PR TITLE
VS Code Overhaul

### LINKED ISSUE #14: VS Code Tutorial duplicate input redirection
There is duplicate input redirection information in the VS Code tutorial [here](https://eecs280staff.github.io/p1-stats/setup_vscode.html#input-redirection) and [here](https://eecs280staff.github.io/p1-stats/setup_vscode.html#troubleshooting--faq).  Remove the second, putting any pitfalls into pro-tip or warning boxes.

### LINKED ISSUE #24: Command line cheat sheet
It would be nice to provide a Linux command line cheat sheet that contains all the commands used in the setup tutorial.

For reference, here's a [past](https://github.com/eecs280staff/eecs280.org/files/5489780/Command_line_tools.pdf) cheat sheet.

## TODO
- [ ] Draft a cheat sheet in Markdown
- [ ] Verify that each command from the setup tutorial appears in the cheat sheet
- [ ] Edit for length, should be 1 page
- [ ] Link from eecs280.org Tutorials section
- [ ] Link from setup tutorial

### LINKED ISSUE #25: Revert Apple Silicon Workaround
In eecs280staff/p1-stats#218 we add a work-around for the Apple M1 architecture.  The official VSCode C/C++ extension does not work on the M1, so we recommend CodeLLDB instead.  Once VSCode's C/C++ extension supports Apple Silicon, we can remove the CodeLLDB stuff.

### PR DESCRIPTION
**Pull request by awdeorio**
Major updates to the VS Code tutorial. 

### Summary of changes
- Include project creation and starter files in this tutorial
- `main.cpp` is generic, not specific to EECS 280 P1
- Provide sample code in Debug section, replacing example that depended on the EECS 280 project 1 file structure.
- Do not provide sample `launch.json`. The default works fine nowadays.
- Prefer to install extensions using `code --install-extension`
- Remove CodeLLDB extension now that Apple Silicon is supported by the official Microsoft C/C++ extension
- Remove Intellisense configuration for C+11. It looks like C++17 is the default now. Replace with a note in trouble shooting.
- Remove CAEN Linux special version loading. A recent version is now the default.

### Related issues
Closes #14
Closes #25
Helps #10
Helps #28

### Validation
[Preview](https://preview.sesh.rs/previews/eecs280staff/tutorials/42/setup_vscode.html)

- [x] Test on macOS / Apple Silicon
- [x] Test on macOS / Intel
- [x] Test on Windows / WSL
- [ ] Test on CAEN Linux

### CODE REVIEW

**Comment by awdeorio** [comment]
@zjweiss would you be willing to test this on Windows?

**Comment by jamesjuett** [comment]
@awdeorio I think some text got unintentionally lost in a merge. Could you take a look at [45050c3](https://github.com/eecs280staff/tutorials/pull/42/commits/45050c3a35a89e5e695e5c3d8c7d81c1c6d3c85c)? Unless you intended to remove it, should be ok now.

File: docs/setup_vscode.md [hunk 1]
```diff
@@ -484,189 +449,163 @@ To configure input redirection, edit `launch.json`.
...
+<img src="images/vscode135.png" width="768px" />
 
-If this method works for you, feel free to continue using the CodeLLDB debugger!
+## Troubleshooting
+This section is for common problems and solutions.
 
-Note that if you are using CodeLLDB, [the above section describing `launch.json`
-configuration](#run) is no longer accurate. Instead, refer to the below
-subsections.
+### Intellisense and C++11
```
Comment (jamesjuett) [inline docs/setup_vscode.md:582]: Change header to C++17 for consistency with the suggested modification? Or just something generic like "Intellisense C++ Standard"?

**Review by jamesjuett — APPROVED** [review]
@awdeorio Made some suggestions - take a look and see what you thing. I'm approving so that I don't block anything, feel free to merge after reviewing my comments and making changes as you think is appropriate. Overall I think this is good to go and additional improvements can be made via future issues.

File: docs/setup_vscode.md [hunk 2]
```diff
@@ -54,26 +61,22 @@ Select "Add to PATH".
 Finally, reboot
```
[the remark this answers is not in the document]
Comment (awdeorio) [reply docs/setup_vscode.md:66]: Done dc191a28391633542c7cde1c7f5280199a392459

File: docs/setup_vscode.md [hunk 3]
```diff
@@ -58,26 +66,22 @@ Select "Add to PATH".
...
+$ code --install-extension ms-vscode-remote.remote-wsl
+$ code --list-extensions
+ms-vscode-remote.remote-wsl
 ```
 
+After the extension is installed, quit VS Code and start it again.
+
 You'll know that VS Code is running in remote mode when you see the remote mode indicator in the bottom left corner.
 
 <img src="images/vscode068.png" width="768px">
```
[the remark this answers is not in the document]
Comment (awdeorio) [reply docs/setup_vscode.md:87]: Great idea.  I added a placeholder but I need a screenshot.  Could you take one for me?  I'm thinking something that looks kinda like [this](https://ljr.im/articles/visual-studio-code-debugging-home-assistant-wsl-version/c5_p1_%E6%89%93%E5%BC%80%E6%9C%AC%E5%9C%B0%E7%9B%AE%E5%BD%95%E5%86%8D%E7%94%A8WSL%E9%87%8D%E6%96%B0%E5%8A%A0%E8%BD%BD.png).  I don't have a Windows machine!

File: docs/setup_vscode.md [hunk 4]
```diff
@@ -86,323 +90,269 @@ You'll know that VS Code is running in remote mode when you see the remote mode
...
+{: data-highlight="2" }
+</div>
 
-## Intellisense
-Intellisense is the feature that indicates compiler errors with red squiggly lines and suggests code completions.  We'll configure the C/C++ Extension's intellisense features to support C++11.
-
-First, you should already have the `C/C++` extension installed ([Instructions](setup_vscode.html#install-the-cc-extension)).
-
-Next, open VS Code's Command Palette with `View > Command Palette` or with the keyboard shortcut `ctrl + shift + P` on Windows or `cmd + shift + P` on macOS.  Search for and select `C/C++: Edit Configurations (JSON)`.  This will open the file `c_cpp_properties.json`.
+Click the triangle to run.  You'll see your program's output in the debug console.
```
[the remark this answers is not in the document]
Comment (awdeorio) [reply docs/setup_vscode.md:353]: Good call.  I added an annotation to the screenshot.

**Review by zjweiss — COMMENTED** [review]
LGTM

Everything was thorough, and I wasn't left with any questions.

**Comment by awdeorio** [comment]
Thanks @zjweiss !

File: docs/setup_vscode.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (awdeorio) [reply docs/setup_vscode.md:100]: I decided to create an issue for this #48

File: docs/setup_vscode.md [hunk 5]
```diff
@@ -412,45 +386,56 @@ Edit the `"environment"` property in your `launch.json`.  If there's already an
...
+Redirect file `main_test.in` to stdin of `main.exe`.
+```console
+$ ./main.exe < main_test.in
+What's your name?
+Hello Drew!
+```
+{: data-highlight="1" }
 
-### Windows `launch.json` changes
+#### Windows `launch.json` changes
```
Comment (ohjuny) [inline docs/setup_vscode.md:438]: No directions for CAEN Linux. After testing, it seems that the Windows changes work for CAEN Linux too.

**Review by ohjuny — COMMENTED** [review]
Just had a run through using CAEN Linux, and everything looks good! Just found one spot where CAEN users may face confusion.

File: docs/setup_vscode.md (hunk 5 above)
Comment (awdeorio) [reply docs/setup_vscode.md:438]: Good catch.  #51

**Pull request merged**